Mathlib/Topology/GPseudoMetric/Basic.lean

```diff
@@ -0,0 +1,328 @@
+/-
+Copyright (c) 2024 Edward van de Meent. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Edward van de Meent
+-/
+import Mathlib.Algebra.CovariantAndContravariant
+import Mathlib.Algebra.NeZero
+import Mathlib.Algebra.Order.Monoid.Lemmas
+import Mathlib.Data.Set.Lattice
+/-!
+# General Pseudo-Metrics
+
+In this file we introduce `GPseudoMetric`.
+This differs from `PseudoMetricSpace` by not requiring that the codomain of the metric be `ℝ`,
+instead only requiring that it is an (additive) commutative monoid, with a linear ordering.
+
+## Main Definitions
+
+- `GPseudoMetric α β`: A structure containing a distance function on `α` with codomain `β`,
```

Comment:
Small grammar comment: in English, you continue in lower-case after a colon. Also in a few other places below.

(Some languages are different. In German, for instance, you use upper-case.)
```suggestion
- `GPseudoMetric α β`: a structure containing a distance function on `α` with codomain `β`,